code/modules/research/xenobiology/crossbreeding/warping.dm

```diff
@@ -0,0 +1,783 @@
+	cooldown = world.time + max_cooldown
+
+/obj/effect/warped_rune/attack_hand(mob/living/user)
+	. = ..()
+	do_effect(user)
+
+/obj/effect/warped_rune/proc/do_effect(mob/user)
+	SHOULD_CALL_PARENT(TRUE)
+	if(deleteme)
+		to_chat(user, ("<span class='notice'>[src] fades.</span>"))
+		qdel(src)
+
+/obj/effect/warped_rune/Crossed(atom/movable/AM, oldloc)
+	SHOULD_CALL_PARENT(TRUE)
+	. = ..()
+	if(deleteme && activated)
+		visible_message("<span class='notice'>[src] fades.</span>")
+		qdel(src)
+
+/obj/item/slimecross/warping/grey
+	name = "greyspace crossbreed"
+	colour = "grey"
+	effect_desc = "Draws a rune. Extracts that are on the rune are absorbed, 8 extracts produces an adult slime of that color."
+	runepath = /obj/effect/warped_rune/greyspace
+
+/obj/effect/warped_rune/greyspace
+	name = "greyspace rune"
+	desc = "Death is merely a setback, anything can be rebuilt given the right components"
+	icon_state = "rune_grey"
+	///extractype is used to remember the type of the extract on the rune
+	var/extractype
+	var/req_extracts = 8
+
+/obj/effect/warped_rune/greyspace/examine(mob/user)
+	. = ..()
+	to_chat(user, "<span class='notice'>Requires absorbing [req_extracts] [extractype ? "[extractype] extracts" : "slime extracts"].</span>")
+
+/obj/effect/warped_rune/greyspace/do_effect(mob/user)
+	for(var/obj/item/slime_extract/extract in rune_turf)
+		if(extract.color_slime == extractype || !extractype) //check if the extract is the first one or of the right color.
+			extractype = extract.color_slime
+			qdel(extract) //vores the slime extract
+			playsound(rune_turf, 'sound/effects/splat.ogg', 20, TRUE)
+			req_extracts--
+			if(req_extracts <= 0)
+				playsound(rune_turf, 'sound/effects/splat.ogg', 20, TRUE)
+				new /mob/living/simple_animal/slime (rune_turf, extractype) //spawn a slime from the extract's color
+				req_extracts = initial(req_extracts)
+				extractype = null // reset extractype to FALSE to allow a new extract type
+				. = ..()
+				break
+		else
+			to_chat(user, "<span class='warning'>Requires a [extractype ? "[extractype] extracts" : "slime extract"].</span>")
+
+
+/obj/item/slimecross/warping/orange
+	colour = "orange"
+	runepath = /obj/effect/warped_rune/orangespace
+	effect_desc = "Draws a rune that can summon a bonfire."
+
+/obj/effect/warped_rune/orangespace
+	desc = "This can be activated to summon a bonfire."
+	icon_state = "rune_orange"
+
+/obj/effect/warped_rune/orangespace/do_effect(mob/user)
+	var/obj/structure/bonfire/bluespace/B = new (rune_turf)
+	B.StartBurning()
+	. = ..()
+
+/obj/item/slimecross/warping/purple
+	colour = "purple"
+	runepath = /obj/effect/warped_rune/purplespace
+	effect_desc = "Draws a rune that may be activated to summon two random medical items "
+
+/obj/effect/warped_rune/purplespace
+	desc = "This can be activated to summon two random medical"
+	icon_state = "rune_purple"
+
+/obj/effect/warped_rune/purplespace/do_effect(mob/user)
+	var/list/medical = list(
```

Comment:
```suggestion
	var/static/list/medical = list(
```

Reply:
why?

Reply:
Statics are essentially a global var exclusive that object _type_. It means you only need one medical list instead of every purplespace run having their own, identical list.

It saves memory

Reply:
yes, but it is a list that is inside a proc, so the list should only be created when it is executed, on the other hand, this proc removes elements from this list

Reply:
Ahhh, it is `pick_n_take`; then static list isn't a very good way, and it should be a new list every call. Not sure why it has to, but, uh, okay.